### PR TITLE
Migrate CSV parsing to league/csv (task #2441)

### PR DESCRIPTION
**Pull request by mamchenkov**
Migrate our own implementation of the CSV parsing to [league/csv](https://packagist.org/packages/league/csv).  As part of this, decouple and refactor the CSV functionality, so that we could get rid of some of the traits.

- [x] Add league/csv to the project
- [x] Add unit tests to compare results from old implementation to the new implementation
- [x] Implement CSV parser classes for migrations, views, lists, etc.
- [x] Replace CsvTrait with CSV parser classes.
- [x] Cleanup
- [x] Test! Test! Test!

### CODE REVIEW

File: src/Parser/Csv/MigrationParser.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    public function wrapFromPath($path, array $structure = [], $wrapField = null)
+    {
+        $result = [];
+
+        // Overwrite default wrap field
+        if (!empty($wrapField)) {
+            $this->wrapField = $wrapField;
+        }
+
```
Comment (flangofas) [inline src/Parser/Csv/MigrationParser.php]: @mamchenkov You need to overwrite $this->sturcture if necessary here?

File: src/Parser/Csv/ViewParser.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * View CSV Parser
+ *
+ * This parser is useful for view CSV processing.
+ * View CSV files can vary in their format, depending
+ * on the need of the view (for example, we currently
+ * support add/edit/view with panels and index view,
+ * which is completely different.
+ *
+ * Therefor, it is assumed that the caller of the
```
Comment (flangofas) [inline src/Parser/Csv/ViewParser.php]: @mamchenkov typo

File: src/Parser/Csv/MigrationParser.php (hunk 1 above)
Comment (mamchenkov) [inline src/Parser/Csv/MigrationParser.php]: @flangofas I'm calling the `parseFromPath()` which does that.

Comment (flangofas) [inline src/Parser/Csv/MigrationParser.php]: @mamchenkov OK :+1:

**Comment by flangofas** [comment]
@mamchenkov Nice! :+1:

**Pull request merged**